### PR TITLE
Improve testing skeleton & fix bugs in Model & TrackIter

### PR DESCRIPTION
**Pull request by UncleGrandpa925**

### CODE REVIEW

File: src/main/java/trackitnus/model/TrackIter.java [hunk 1]
```diff
@@ -249,10 +251,6 @@ public void removeLesson(Lesson key) {
         lessons.remove(key);
     }
 
-    public void sortLesson() {
-        lessons.sort();
-    }
-
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/TrackIter.java:255]: Why is this removed? Is it because not called at all? Minh wanted to keep this in initially so maybe should discuss w him before removing it

File: src/main/java/trackitnus/model/lesson/DayOfWeek.java [hunk 2]
```diff
@@ -13,8 +12,8 @@
     Sat;
     private static final int SIZE = 7;
 
-    public static LessonWeekday getLessonWeekDay(LocalDate date) {
-        DayOfWeek dayOfWeek = date.getDayOfWeek();
+    public static DayOfWeek getLessonWeekDay(LocalDate date) {
+        java.time.DayOfWeek dayOfWeek = date.getDayOfWeek();
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/lesson/DayOfWeek.java:16]: Wouldn't it be better to keep the import so you dont need to type java.time.DayOfWeek

File: src/main/java/trackitnus/model/lesson/UniqueLessonList.java [hunk 3]
```diff
@@ -102,7 +102,6 @@ public void setLessons(List<Lesson> lessons) {
      * Returns the backing list as an unmodifiable {@code ObservableList}.
      */
     public ObservableList<Lesson> asUnmodifiableObservableList() {
-        sort();
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/lesson/UniqueLessonList.java:105]: If you remove this, how will we ensure that Lessons are always sorted?

File: src/main/java/trackitnus/model/task/UniqueTaskList.java [hunk 4]
```diff
@@ -102,7 +102,6 @@ public void setTasks(List<Task> tasks) {
      * Returns the backing list as an unmodifiable {@code ObservableList}.
      */
     public ObservableList<Task> asUnmodifiableObservableList() {
-        internalList.sort(new TaskComparator());
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/task/UniqueTaskList.java:105]: If you remove this, how will we ensure that Tasks are always sorted?

**Review by simonteozw — COMMENTED** [review]
Looks good. Just a few comments

File: src/main/java/trackitnus/model/TrackIter.java (hunk 1 above)
Comment (UncleGrandpa925) [reply src/main/java/trackitnus/model/TrackIter.java:255]: It has been moved haha

**Review by simonteozw — APPROVED** [review]
LGTM
